realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -381,6 +381,22 @@ public int size() {
         }
     }
 
+    @Override
+    public boolean contains(Object object) {
+        boolean contains = false;
+        if (object == null) {
+            contains = false;
+        } else if (RealmObject.class.isAssignableFrom(object.getClass())) {
```

Comment:
`if (object.getClass().isAssignableFrom(clazz.getClass()))`

What about this? To make it safer? Row could be same for different `RealmObject`.